go/internal/utils/buf_reader.go

```diff
@@ -71,16 +78,17 @@ func (b *bufferedReader) fill() {
 	}
 
 	if b.w >= len(b.buf) {
-		panic("parquet/bufio: tried to fill full buffer")
+		return fmt.Errorf("arrow/bufferedreader: %w", bufio.ErrBufferFull)
```

Comment:
You use errors.New and fmt.Errorf through-out the file.
Is it worth consolidating or are they offering diff functionality?

Reply:
they are different functionality. `errors.New` creates a new error from a single string. I'm using `fmt.Errorf` when I'm wrapping an error so that callers can still use `errors.Is(err, bufio.ErrBufferFull)` or otherwise to handle the errors